ESP8266/ESP8266.cpp

```diff
@@ -70,6 +80,43 @@ int ESP8266::get_firmware_version()
     }
 }
 
+bool ESP8266::start_uart_hw_flow_ctrl(void)
+{
+    bool done = true;
+    static const int DATABITS_8 = 8;  // 8-bit data
+    static const int STOPBITS_1 = 1;  // 1-bit stop bit
+    static const int PARITY_NONE = 0; // None
+    static const int FC_RTS = 1;      // Flow Control RTS
+    static const int FC_CTS = 2;      // Flow Control CTS
+    static const int FC_RTSCTS = 3;   // Flow Control RTS&CTS
+
+    if (_serial_rts != NC && _serial_cts != NC) {
+        // Start board's flow control
+        _serial.set_flow_control(SerialBase::RTSCTS, _serial_rts, _serial_cts);
+
+        // Start ESP8266's flow control
+        done = _parser.send("AT+UART_CUR=%u,%u,%u,%u,%u", ESP8266_DEFAULT_BAUD_RATE, DATABITS_8, STOPBITS_1,
+                PARITY_NONE, FC_RTSCTS)
+                        && _parser.recv("OK\n");
+    } else if (_serial_rts != NC) {
+        _serial.set_flow_control(SerialBase::RTS, _serial_rts, NC);
+
+        // Start ESP8266's flow control
+        done = _parser.send("AT+UART_CUR=%u,%u,%u,%u,%u", ESP8266_DEFAULT_BAUD_RATE, DATABITS_8, STOPBITS_1,
+                PARITY_NONE, FC_CTS)
+                        && _parser.recv("OK\n");
+    } else if (_serial_cts != NC) {
+        // Start ESP8266's flow control
+        done = _parser.send("AT+UART_CUR=%u,%u,%u,%u,%u", ESP8266_DEFAULT_BAUD_RATE, DATABITS_8, STOPBITS_1,
+                PARITY_NONE, FC_RTS)
+                        && _parser.recv("OK\n");
+
+        _serial.set_flow_control(SerialBase::CTS, NC, _serial_cts);
+    }
```

Comment:
Where is the `else` case?
When no flow control pins are enabled, should it still call something, or is the point entirely skip it?

If device have flow control set as a default, should we still disable it?

Reply:
Also, on last case `_serial.set_flow_contro()` is called after `_parser.send()`, is this intentional?

Reply:
I understad it so that we are not taking orders from ESP8266 when to send until ESP8266 is configured to do so.

Reply:
When there are no pins configured for flow control there is no need to make any settings